src/features/ci-cd/hooks/track-latest-release.ts

```diff
@@ -1,138 +1,165 @@
 import * as _ from 'lodash';
+import { sbvrUtils, hooks } from '@balena/pinejs';
 
-import { hooks } from '@balena/pinejs';
 const releaseStateFields = [
 	'status',
 	'is_passing_tests',
 	'release_type',
 	'is_invalidated',
 ];
-const updateLatestRelease = async ({ request, api }: hooks.HookArgs) => {
-	if (
-		// we avoid short-circuiting for POSTs since the db can provide defaults that we later use to filter on.
-		request.method !== 'POST' &&
-		!releaseStateFields.some((field) => field in request.values)
-	) {
+
+const updateLatestRelease = async (
+	api: sbvrUtils.PinejsClient,
+	releaseIds: number[],
+) => {
+	if (!releaseIds.length) {
 		return;
 	}
-	const [release] = await api.get({
-		resource: 'release',
+
+	const appsToUpdate = await api.get({
+		resource: 'application',
 		options: {
 			$select: 'id',
-			$top: 1,
-			// the most recently started build should be the latest, to ignore variable build times
-			$orderby: { start_timestamp: 'desc' },
-			$filter: {
-				// We only track releases that are successful, final, not invalid, and passing tests
-				release_type: 'final',
-				is_passing_tests: true,
-				is_invalidated: false,
-				status: 'success',
-			},
 			$expand: {
-				belongs_to__application: {
+				owns__device: {
 					$select: ['id'],
-					$expand: {
-						owns__device: {
-							$select: ['id'],
-						},
-					},
 				},
-				contains__image: {
-					$select: ['id'],
+				owns__release: {
+					$select: 'id',
+					$top: 1,
+					// the most recently started build should be the latest, to ignore variable build times
+					$orderby: { start_timestamp: 'desc' },
+					$filter: {
+						// We only track releases that are successful, final, not invalid, and passing tests
+						release_type: 'final',
+						is_passing_tests: true,
+						is_invalidated: false,
+						status: 'success',
+					},
 					$expand: {
-						image: {
+						contains__image: {
 							$select: ['id'],
 							$expand: {
-								is_a_build_of__service: {
+								image: {
 									$select: ['id'],
+									$expand: {
+										is_a_build_of__service: {
+											$select: ['id'],
+										},
+									},
 								},
 							},
 						},
 					},
 				},
 			},
-		},
-	});
-	if (release == null) {
-		return;
-	}
-	await api.patch({
-		resource: 'application',
-		id: release.belongs_to__application[0].id,
-		options: {
 			$filter: {
-				should_track_latest_release: true,
+				owns__release: {
```

Comment:
In the rat I was originally also filtering on `should_be_running__release: true`, which while writing the tests I found that it would break new service install creation for pinned apps.